src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,69 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SUPPLY_VALUE = "supply";
+    private static final String BUY_VALUE = "buy";
+    private static final String RESULT = "result";
+    private static final String SEPARATOR = ",";
+    private static final int INDEX_ZERO = 0;
+    private static final int INDEX_ONE = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String[] splitFile = read(fromFileName).split(System.lineSeparator());
+        int supplyCount = 0;
+        int buyCount = 0;
+        int result;
+
+        for (String eachElement : splitFile) {
+            String[] toCountElement = eachElement.split(SEPARATOR);
+            String type = toCountElement[INDEX_ZERO];
+            int count = Integer.parseInt((toCountElement[INDEX_ONE]));
+            if (type.equals(SUPPLY_VALUE)) {
+                supplyCount += count;
+            }
+            if (type.equals(BUY_VALUE)) {
+                buyCount += count;
+            }
+        }
+        result = supplyCount - buyCount;
+        String finalDataFile = dataCreator(supplyCount, buyCount, result);
+        write(toFileName, finalDataFile);
+    }
+
+    private String dataCreator(int supplyCount, int buyCount, int result) {
```

Comment:
dataCreator name is suitable for class but not for the method that returns String value
```suggestion
    private String generateReport(int supplyCount, int buyCount, int result) {
```